src/context/WidgetSettingContext.tsx

```diff
@@ -195,6 +200,7 @@ export const WidgetSettingProvider = ({
         },
         widgetSession,
         initManualSession,
+        resetSession: () => initSessionByStrategy(sessionStrategy, true),
```

Comment:
Shouldn't we be handling error here? onChannelFetchFailed is called by UIKit but initSessionByStrategy does not fetch channel through UIKit so failing here would not automatically call onChannelFetchFailed

Reply:
To do so, we need to make an unnecessary API call just to check if the channel exists.

Reply:
ok yes but my point is not about GET or POST but about synchronizing the retries. Maybe I misunderstood but I thought

If uikit fails fetching channel, onChannelFetchFailed() will be called which then calls resetSession(). This tries to create channel. What happens if this fails?

Reply:
seeing resetReqCounter, I expected onChannelFetchFailed would be called over and over again upto five times, is resetSession() failure going to call onChannelFetchFailed ?

Reply:
> ok yes but my point is not about GET or POST but about synchronizing the retries. Maybe I misunderstood but I thought
> 
> If uikit fails fetching channel, onChannelFetchFailed() will be called which then calls resetSession(). This tries to create channel. What happens if this fails?

If it fails, nothing happens.

Reply:
> seeing resetReqCounter, I expected onChannelFetchFailed would be called over and over again upto five times, is resetSession() failure going to call onChannelFetchFailed ? bc I could not find connection between the two

`resetSession` is a method that creates a new channel, and `onChannelFetchFailed` is a method that gets called when fetching the channel fails.

There is no direct connection between `resetSession` and `onChannelFetchFailed`, but the logic flow is cyclic.
>fetch channel -> not found channel -> `onChannelFetchFailed` -> `resetSession` -> channel created -> fetch channel -> good !

If the channel gets deleted in the middle of this flow, an infinite loop can occur.
>fetch channel -> not found channel -> `onChannelFetchFailed` -> `resetSession` -> channel created -> **channel deleted** -> fetch channel -> not found channel -> `onChannelFetchFailed` -> ... loop

I believe the probability of this happening in actual business logic is close to zero.
Nevertheless, since this logic has the potential to cause an infinite loop, this is a defensive code to prevent it.